os/fs/smartfs/smartfs_smart.c

```diff
@@ -1954,15 +1954,22 @@ int smartfs_rename(struct inode *mountpt, const char *oldrelpath, const char *ne
 
 	ret = smartfs_finddirentry(fs, &newentry, newrelpath);
 	if (ret == OK) {
-		/* Test if it's a file.  If it is, then it's an error */
-
-		if ((newentry.flags & SMARTFS_DIRENT_TYPE) == SMARTFS_DIRENT_TYPE_FILE) {
+		/* If the old entry and the new entry found are of the same type, then rename isn't possible as the entry already exists */
+		if ((oldentry.flags & SMARTFS_DIRENT_TYPE) == (newentry.flags & SMARTFS_DIRENT_TYPE)) {
```

Comment:
We don't have to check type, if path is same it shoud return error.

Reply:
Will change if we let go of move functionality

Reply:
Will update line 2027 accordingly too